### PR TITLE
docs: add getting-started tutorial for new contributors

### LINKED ISSUE #461: docs(documentation): create getting-started tutorial and external deps guide
README stops at `make test` — there's no guidance on training, data generation, or evaluation. External dependencies (Surge XT, rclone, R2, RunPod, W&B) are mentioned in code but not in any setup guide. Newcomers discover requirements by hitting errors.

## Scope

Create `docs/getting-started.md` with:

### 1. k-osc quickstart (no external deps)
- Run: `python src/train.py +experiment=kosc/baseline trainer.max_epochs=5`
- Show expected terminal output
- Where checkpoints land
- Optional: W&B dashboard setup (link to `docs/reference/wandb-integration.md`)

### 2. Hydra config override examples
- How to change model: `+experiment=kosc/other_variant`
- How to override batch size: `data.batch_size=32`
- How to override learning rate: `model.lr=1e-4`
- Reference: `docs/reference/configuration-reference.md`

### 3. External dependency setup (each with install + verify steps)

**Surge XT VST plugin:**
- What it is: open-source synthesizer used for dataset generation
- Download/install instructions
- Verify: `make test -m requires_vst` (or specific VST test)

**rclone + Cloudflare R2:**
- Install rclone
- Configure R2 remote with credentials
- Verify: `rclone ls r2:intermediate-data/` (or similar)

**Weights & Biases:**
- Get API key, run `wandb login`
- Link to existing `docs/reference/wandb-integration.md`

**RunPod:**
- When you need it (distributed data generation only)
- Account setup, API key
- Not needed for local development

### 4. What to try next
- Experiment configs in `configs/experiment/`
- Evaluation: `python src/eval.py`
- Data generation: `pipeline/entrypoints/generate_dataset.py`

## Audit items bundled
- Audit item 7: No "first run" tutorial
- Audit item 8: External deps undocumented

Part of #457

### PR DESCRIPTION
**Pull request by ktinubu**
## Summary
- Add `docs/getting-started.md` with step-by-step setup tutorial for new users
- Covers prerequisites, installation, pre-commit setup, and virtual environment creation
- Includes k-osc quickstart (no external deps needed) with working training commands
- Documents external dependency setup: Surge XT, rclone/R2, W&B, RunPod
- Covers Hydra config system with common override examples
- Adds evaluation, Docker workflow, and troubleshooting sections

Closes #461
Part of #457

### CODE REVIEW

**Comment by ktinubu** [comment]
## Verification Results

### Level 1: End-to-end training with documented command

- [ ] **Quickstart training command runs to completion** *(Level 1 -- SKIP: CPU training with default dataset size of 409M samples exceeds environment timeout. The config resolves correctly (verified at Level 2 below), and training initializes, but data generation is too slow on CPU for a timed verification.)*
  To verify manually:
  ```bash
  # Use a smaller dataset for quick smoke test:
  python src/train.py experiment=kosc/ffn_mse trainer.max_steps=100 trainer=cpu logger=csv test=False data.train_val_test_sizes='[1024, 256, 256]'
  # Or on GPU (much faster):
  python src/train.py experiment=kosc/ffn_mse trainer.max_steps=100 logger=csv test=False
  ```

### Level 2: Documented commands resolve valid Hydra configs

- [x] **Quickstart training command resolves a valid config** *(Level 2 -- Hydra `--cfg job` proves the documented command composes a valid config without errors)*
  ```bash
  $ python src/train.py experiment=kosc/ffn_mse trainer.max_steps=5000 trainer=cpu logger=csv test=False --cfg job 2>&1 | tail -5
  experiment_name: kosc
  run_name: ffn_mse
  ```

- [x] **batch_size override works** *(Level 2 -- Hydra config resolution)*
  ```bash
  $ python src/train.py experiment=kosc/ffn_mse data.batch_size=32 trainer=cpu logger=csv test=False --cfg job 2>&1 | grep batch_size
    batch_size: 32
  ```

- [x] **model.optimizer.lr override works** *(Level 2 -- Hydra config resolution)*
  ```bash
  $ python src/train.py experiment=kosc/ffn_mse model.optimizer.lr=1e-4 trainer=cpu logger=csv test=False --cfg job 2>&1 | grep "lr:"
      lr: 0.0001
  ```

- [x] **logger=tensorboard override works** *(Level 2 -- Hydra config resolution)*
  ```bash
  $ python src/train.py experiment=kosc/ffn_mse logger=tensorboard trainer=cpu test=False --cfg job 2>&1 | grep "tensorboard" | head -2
    tensorboard:
      _target_: lightning.pytorch.loggers.tensorboard.TensorBoardLogger
  ```

### Level 2: Referenced file paths exist

- [x] **All 37 referenced file/directory paths exist in the repo** *(Level 2 -- direct filesystem check of every path mentioned in the tutorial)*
  ```bash
  $ python3 -c "
  import os, sys
  paths = [
      'src/train.py', 'src/eval.py', 'configs/train.yaml', 'configs/eval.yaml',
      'configs/data', 'configs/model', 'configs/trainer', 'configs/logger',
      'configs/callbacks', 'configs/experiment', 'configs/experiment/kosc',
      'configs/experiment/surge', 'configs/dataset',
      'configs/experiment/kosc/base.yaml', 'configs/experiment/kosc/ffn_mse.yaml',
      'configs/experiment/kosc/ffn_chamfer.yaml', 'configs/experiment/kosc/flow.yaml',
      'configs/experiment/kosc/flow_asym.yaml', 'configs/experiment/kosc/flowmlp.yaml',
      'configs/logger/wandb.yaml', 'configs/logger/csv.yaml',
      'configs/logger/tensorboard.yaml', 'configs/trainer/cpu.yaml',
      'configs/trainer/gpu.yaml', 'configs/trainer/mps.yaml',
      'configs/trainer/ddp.yaml', 'configs/debug',
      'pipeline/entrypoints/generate_dataset.py',
      'docs/design/data-pipeline.md', 'docs/design/training-pipeline.md',
      'docs/reference/wandb-integration.md', 'docs/reference/configuration-reference.md',
      'docs/reference/docker.md', 'requirements.txt',
      'docker/ubuntu22_04/Dockerfile', 'Makefile', '.pre-commit-config.yaml',
  ]
  missing = [p for p in paths if not os.path.exists(p)]
  if missing:
      print(f'MISSING: {missing}'); sys.exit(1)
  print(f'All {len(paths)} referenced paths exist.')
  "
  All 37 referenced paths exist.
  ```

### Level 2: Internal doc links resolve

- [x] **Relative links from docs/ resolve to existing files** *(Level 2 -- filesystem check of doc-relative links)*
  ```bash
  $ python3 -c "
  import os
  links = ['reference/wandb-integration.md', 'reference/configuration-reference.md', 'reference/docker.md']
  for link in links:
      path = os.path.join('docs', link)
      print(f\"{'OK' if os.path.exists(path) else 'MISSING'}: {link}\")
  "
  OK: reference/wandb-integration.md
  OK: reference/configuration-reference.md
  OK: reference/docker.md
  ```

### Level 2: Make targets exist

- [x] **All referenced make targets exist in Makefile** *(Level 2 -- pattern match against Makefile targets)*
  ```bash
  $ python3 -c "
  import re
  with open('Makefile') as f: makefile = f.read()
  for t in ['test', 'format', 'help', 'docker-build-dev-snapshot']:
      found = bool(re.search(rf'^{re.escape(t)}:', makefile, re.MULTILINE))
      print(f\"{'OK' if found else 'MISSING'}: make {t}\")
  "
  OK: make test
  OK: make format
  OK: make help
  OK: make docker-build-dev-snapshot
  ```

### Adversarial: Config table accuracy

- [x] **kosc experiment configs listed in tutorial all exist on disk** *(Level 2 -- cross-reference tutorial table against actual directory contents)*
  ```bash
  $ python3 -c "
  import os
  listed = {'base', 'ffn_mse', 'ffn_chamfer', 'flow', 'flow_asym', 'flowmlp'}
  actual = {f.replace('.yaml', '') for f in os.listdir('configs/experiment/kosc') if f.endswith('.yaml')}
  missing = listed - actual
  unlisted = actual - listed
  print(f'Listed: {sorted(listed)}')
  print(f'Actual: {sorted(actual)}')
  print(f'Missing from dir: {sorted(missing)}' if missing else 'All listed configs exist.')
  print(f'Unlisted (not an error): {sorted(unlisted)}' if unlisted else '')
  "
  Listed: ['base', 'ffn_chamfer', 'ffn_mse', 'flow', 'flow_asym', 'flowmlp']
  Actual: ['base', 'ffn_asym', 'ffn_chamfer', 'ffn_chamfer_asym', 'ffn_mse', 'ffn_sort', 'flow', 'flow_asym', 'flow_fixed', 'flow_fixed_asym', 'flowmlp', 'flowmlp_asym']
  All listed configs exist.
  Unlisted (not an error): ['ffn_asym', 'ffn_chamfer_asym', 'ffn_sort', 'flow_fixed', 'flow_fixed_asym', 'flowmlp_asym']
  ```
  Tutorial shows a representative subset (6 of 12). All listed configs exist. Unlisted configs are advanced variants.

### Adversarial: Stale or incorrect Hydra syntax

- [x] **Tutorial uses correct `experiment=` syntax (not `+experiment=`)** *(Level 2 -- original draft used `+experiment=` which fails because `train.yaml` already defines `experiment: null`; caught and fixed during verification)*
  ```bash
  $ python src/train.py experiment=kosc/ffn_mse trainer=cpu logger=csv test=False --cfg job 2>&1 | grep "experiment_name"
  experiment_name: kosc
  ```
  Fix committed in `2eda312`.

### Adversarial: Tutorial claims `trainer.max_epochs` override

- [x] **Tutorial uses `trainer.max_steps` (not `trainer.max_epochs`)** *(Level 2 -- original draft used `max_epochs` which does not exist in the trainer config; caught and fixed during verification)*
  ```bash
  $ grep "max_steps\|max_epochs" configs/trainer/default.yaml
  min_steps: 400_000
  max_steps: 400_000
  ```
  Fix committed in `2eda312`.

File: docs/getting-started.md [hunk 1]
```diff
@@ -0,0 +1,415 @@
...
+
+The k-osc task is a synthetic benchmark where the model learns to predict
+parameters of a sum-of-sinusoids signal. It generates data on the fly, so you
+do not need any external datasets, VST plugins, or cloud storage.
+
+### 3a. Train a model
+
+```bash
+python src/train.py experiment=kosc/ffn_mse trainer.max_steps=5000
+```
```
[the remark this answers is not in the document]
Comment (ktinubu) [reply docs/getting-started.md:100]: The `+experiment=` syntax was already fixed in 2eda312. No further change needed for this item.

File: docs/getting-started.md [hunk 2]
```diff
@@ -0,0 +1,415 @@
...
+
+This runs a feed-forward network with MSE loss on the k-osc task for 5,000
```
[the remark this answers is not in the document]
Comment (ktinubu) [reply docs/getting-started.md:109]: Fixed in fe60ea7 — added `trainer.min_steps=null` override to the quickstart command so the run actually stops at 5,000 steps (the default `min_steps: 400_000` in `configs/trainer/default.yaml` would otherwise prevent early stopping). Also added an explanatory note.

File: docs/getting-started.md [hunk 3]
```diff
@@ -0,0 +1,415 @@
...
+training steps. You should see Lightning's progress bar with decreasing loss
+values.
+
+**What happens:**
+
+- Hydra composes the config from `configs/train.yaml` + the experiment override
+- Lightning sets up the data module, model, callbacks, and trainer
+- Checkpoints are saved under `logs/train/runs/<timestamp>/checkpoints/`
```
[the remark this answers is not in the document]
Comment (ktinubu) [reply docs/getting-started.md]: Fixed in fe60ea7 — updated the checkpoint path to match the Hydra output dir pattern from `configs/hydra/default.yaml`: `logs/{task_name}/{experiment_name}/{run_name}-{timestamp}/checkpoints/`, with a concrete example for the kosc/ffn_mse experiment.

File: docs/getting-started.md [hunk 4]
```diff
@@ -0,0 +1,415 @@
...
+## 6. Evaluation
+
+After training, evaluate the model on the test set:
+
+```bash
+python src/eval.py
+```
+
+By default, evaluation uses the config and checkpoint from the most recent
+training run. You can point it at a specific checkpoint:
```
[the remark this answers is not in the document]
Comment (ktinubu) [reply docs/getting-started.md]: Fixed in fe60ea7 — removed the incorrect claim about defaulting to the most recent run. The eval section now states that `ckpt_path` is required and shows it as a mandatory argument, consistent with `configs/eval.yaml` (`ckpt_path: ???`) and the assertion in `src/eval.py`.

File: docs/getting-started.md [hunk 5]
```diff
@@ -0,0 +1,415 @@
...
+R2_BUCKET=<bucket-name>
+```
+
+**Verify:**
+
+```bash
+rclone ls r2:<bucket-name>/ --max-depth 1
+```
+
+You should see top-level directories like `data/` and `metadata/`.
```
[the remark this answers is not in the document]
Comment (ktinubu) [reply docs/getting-started.md]: Fixed in fe60ea7 — switched from `rclone ls` to `rclone lsd` (list directories), removed the incorrect `--max-depth 1` flag, and corrected the expected top-level directories to `data/`, `train/`, and `eval/` per `docs/design/storage-provenance-spec.md`.

File: docs/getting-started.md [hunk 6]
```diff
@@ -0,0 +1,415 @@
...
+**Build the image:**
+
+```bash
+make docker-build-dev-snapshot \
+  GIT_REF=$(git rev-parse HEAD) \
+  GIT_PAT=<your-github-pat> \
+  R2_ACCESS_KEY_ID=<key> \
+  R2_SECRET_ACCESS_KEY=<secret> \
+  R2_ENDPOINT=<endpoint> \
+  R2_BUCKET=<bucket>
```
[the remark this answers is not in the document]
Comment (ktinubu) [reply docs/getting-started.md]: Fixed in fe60ea7 — added `DOCKER_BUILD_FLAGS=--load` to the Docker build example, consistent with `docs/reference/docker.md`.

File: docs/getting-started.md [hunk 7]
```diff
@@ -0,0 +1,415 @@
...
+git clone --recurse-submodules https://github.com/tinaudio/synth-setter.git
+cd synth-setter
+```
+
+If you already cloned without `--recurse-submodules`, initialize the submodules:
+
+```bash
+git submodule update --init
+```
+
```
[the remark this answers is not in the document]
Comment (ktinubu) [reply docs/getting-started.md:33]: Fixed in 43669ec. Added a blockquote note after the clone instructions explaining the SSH submodule URL and the `git config url."https://github.com/".insteadOf git@github.com:` workaround.

File: docs/getting-started.md [hunk 8]
```diff
@@ -0,0 +1,415 @@
...
+Follow the prompts to create a new remote named `r2` with provider
+`Cloudflare R2` (or `S3` with the R2 endpoint). Alternatively, set these
+environment variables in your `.env` file:
+
+```
+R2_ACCESS_KEY_ID=<your-access-key>
+R2_SECRET_ACCESS_KEY=<your-secret-key>
+R2_ENDPOINT=<your-r2-endpoint-url>
```
[the remark this answers is not in the document]
Comment (ktinubu) [reply docs/getting-started.md:219]: Fixed in 43669ec. Switched to `RCLONE_CONFIG_R2_*` names (including `TYPE` and `PROVIDER`) for the local rclone auto-configuration block, and added a separate block with the `R2_*` names used by the Docker BuildKit secrets, with a cross-reference to section 7.

File: docs/getting-started.md [hunk 9]
```diff
@@ -0,0 +1,415 @@
...
+**Build the image:**
+
+```bash
+make docker-build-dev-snapshot \
+  GIT_REF=$(git rev-parse HEAD) \
+  GIT_PAT=<your-github-pat> \
+  R2_ACCESS_KEY_ID=<key> \
+  R2_SECRET_ACCESS_KEY=<secret> \
+  R2_ENDPOINT=<endpoint> \
+  R2_BUCKET=<bucket> \
```
[the remark this answers is not in the document]
Comment (ktinubu) [reply docs/getting-started.md]: Fixed in 43669ec. Replaced the inline credentials with `set -a && source .env && set +a` before the `make` call, and added a note listing which env vars the target expects to find in `.env`.

File: docs/getting-started.md [hunk 10]
```diff
@@ -0,0 +1,447 @@
...
+
+The k-osc task is a synthetic benchmark where the model learns to predict
+parameters of a sum-of-sinusoids signal. It generates data on the fly, so you
+do not need any external datasets, VST plugins, or cloud storage.
+
+### 3a. Train a model
+
+```bash
+python src/train.py experiment=kosc/ffn_mse trainer.max_steps=5000 trainer.min_steps=null
+```
```
[the remark this answers is not in the document]
Comment (ktinubu) [reply docs/getting-started.md:100]: Fixed in 8bc101c -- added a callout with `trainer=cpu` and `trainer=mps` alternatives right below the quickstart command.

File: docs/getting-started.md [hunk 11]
```diff
@@ -0,0 +1,447 @@
...
+
+### 3b. Available k-osc experiments
+
+The `configs/experiment/kosc/` directory contains several variants:
+
+| Config             | Description                          |
+| ------------------ | ------------------------------------ |
+| `kosc/base`        | Base config (used by other variants) |
+| `kosc/ffn_mse`     | Feed-forward network, MSE loss       |
+| `kosc/ffn_chamfer` | Feed-forward network, Chamfer loss   |
```
[the remark this answers is not in the document]
Comment (ktinubu) [reply docs/getting-started.md:131]: The table already uses standard single-pipe GFM syntax (`| Config | Description |`). Verified it renders correctly -- no change needed.

File: docs/getting-started.md [hunk 12]
```diff
@@ -0,0 +1,447 @@
...
+A Dockerfile is provided for reproducible environments (training, CI, cloud
+deployment). The image bakes in the source code, dependencies, Surge XT, and
+rclone/R2 configuration.
+
+**Build the image:**
+
+The Makefile reads secrets from environment variables and passes them as
+BuildKit secrets (never embedded in image layers). Load your `.env` first to
+avoid leaking credentials in shell history or process listings:
+
```
[the remark this answers is not in the document]
Comment (ktinubu) [reply docs/getting-started.md]: Fixed in 8bc101c -- added a warning callout that built images contain credentials in the filesystem and should not be pushed to public registries.

File: docs/getting-started.md [hunk 13]
```diff
@@ -0,0 +1,459 @@
...
+3. Note the installation path (the test suite needs to find the plugin binary)
+
+**Verify:**
+
+```bash
+pytest -m requires_vst -v
+```
+
+If the plugin is found, VST-dependent tests will run. If not, they are
+automatically skipped (they are excluded from `make test`).
```
[the remark this answers is not in the document]
Comment (ktinubu) [reply docs/getting-started.md:169]: tracked in issue
